src/extension.ts

```diff
@@ -756,18 +756,25 @@ export class Extension implements RunHooks {
 
   private _showTrace(testItem: vscodeTypes.TestItem) {
     const traceUrl = (testItem as any)[traceUrlSymbol];
-    const testModel = this._models.selectedModel();
-    if (testModel)
-      this._traceViewer.open(traceUrl, testModel.config);
+    if (traceUrl)
+      this._enabledTraceView()?.open(traceUrl);
   }
 
   private _treeItemSelected(treeItem: vscodeTypes.TreeItem | null) {
     if (!treeItem)
       return;
     const traceUrl = (treeItem as any)[traceUrlSymbol] || '';
-    const testModel = this._models.selectedModel();
-    if (testModel)
-      this._traceViewer.open(traceUrl, testModel.config);
+    if (!traceUrl && !this._currentTraceViewer?.isStarted())
+      return;
+    this._enabledTraceView()?.open(traceUrl);
+  }
+
+  private _enabledTraceView() {
+    const traceViewer = this._models.selectedModel()?.enabledTraceViewer();
+    if (traceViewer !== this._currentTraceViewer)
+      this._currentTraceViewer?.close();
```

Comment:
I wasn't expecting enabledTraceViewer to do something. Also the naming is probably suboptimal. If `enabledTraceViewer()` "closes" trace viewers, we'll run of trace viewers. Does it mean to say closeTrace?

Reply:
Ok, I understand. Instead of closing trace viewer here, I'll make sure trace viewers corresponding to unselected testModels are closed.

Addressed in https://github.com/microsoft/playwright-vscode/pull/509/commits/643d3c37a4d9c71e7d12ecb2e56016e98feccfe9.